src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray;n_loc_int::Int64=5)
+    location_selection_frequencies(inv_log::NamedDimsArray{D,T,4,A};dims::Union{Symbol,Vector{Symbol}}=:coral_id) where {D,T,A}
+
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_loc_int` : number of locations which are intervened at for each intervention decision.
+- `dims` : dimensions to sum selection frequencies over.
+
+# Returns 
+Counts for location selection at each location in the domain.
+"""
+function location_selection_frequencies(
+    ranks::NamedDimsArray;
+    n_loc_int::Int64=5,
+)
+    ranks_frequencies = ranks_to_frequencies(ranks; n_ranks=n_loc_int)
+    loc_count = sum(ranks_frequencies[ranks=1:n_loc_int], dims=2)[ranks=1]
+
+    return loc_count
+end
+function location_selection_frequencies(
+    inv_log::NamedDimsArray{D,T,4,A};
+    dims::Union{Symbol,Vector{Symbol}}=:coral_id,
+) where {D,T,A}
+    loc_count = dropdims(
+        sum(dropdims(sum(inv_log; dims=dims); dims=dims) .> 0; dims=:scenarios);
+        dims=:scenarios,
+    )
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,3,A};dims::Union{Symbol,Vector{Symbol}}=[:scenarios, :timsteps],
+            ) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,2,A};) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray,dims::Union{Symbol,Vector{Symbol}},)
+
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `dims` : Dimensions to sum over.
+
+# Returns 
+Inverse rankings (i.e. the greater the number the higher ranked the site).
+"""
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,3,A};
+    dims::Vector{Symbol}=[:scenarios, :timesteps],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,2,A};
+    dims::Vector{Symbol}=[:scenarios],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray,
+    dims::Vector{Symbol},
+)
+    n_ranks = maximum(ranks)
+    inv_ranks = dropdims(sum(n_ranks .- ranks; dims=dims); dims=dims[1])
+    return inv_ranks ./ (n_ranks * prod([size(ranks, d) for d in dims]))
```

Comment:
Also: what if instead of normalizing by `(n_ranks * prod([size(ranks, d) for d in dims]))` we normalize by `maximum(score)`? Then we would have score going from 0 to 1, with 0 being the worst and 1 the best. This might be easier to interpret in the future, what do you think?

Reply:
Because we sum over dims, normalising by dividing by maximum(rank)*length(dim1)*length(dim2)... makes sure the values are between 0 and 1. This is because, for example, MAX(dropdims(sum(n_ranks .- ranks; dims=:scenarios); dims=:scenarios))) = n_ranks* (number of scenarios). Alternatively `inv_ranks = dropdims(sum((n_ranks .- ranks)/n_ranks; dims=dims); dims=dims[1])` should give the same result. Is that a clearer way of doing it?

Reply:
Hum, two things:

1 - I might be doing some wrong math here, but shouldn't it be `(maximum(rank)-1)*length(dim1)*etc` ? Suppose you have 4 locations and 3 timesteps. The maximum score would be 4. The best case would be for a location to have ranks [1, 1, 1] (is that right? one for each `timestep`?), which would give a non-normalized score of 9 (and not 4*3 = 12).

2 - In any case, there's one question to consider: if in a particular run there was NO LOCATION with maximum possible score, than there will be no location with normalized score 1. I think that's the difference between doing what you are proposing and normalizig by "the greatest score". In the second option we ALWAYS will have some location with score 1. In the first option, not always. (I'm not saying we should do one or the other, just that this is the difference). @ConnectedSystems what do you think?

Reply:
I'll have to look at this more closely later, sorry, as I've lost context. Need to understand why `prod`...

Reply:
To answer your questions @Zapiano , you're correct it should be -1. We could use the number of sites +1 as the set maximum value to avoid the case where the maximum rank changes. In this case it would be (n_sites*length(dim1)*length(dim2)... etc. I would prefer this as we could compare easily between runs. Let me know what you think.